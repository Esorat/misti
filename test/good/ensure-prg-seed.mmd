graph TD
subgraph C__test
    C__test_141["return nativeRandom()"]:::exitNode
end
