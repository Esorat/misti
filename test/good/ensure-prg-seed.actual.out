[MEDIUM] EnsurePrgSeed: PRG seed should be initialized before using nativeRandom
test/good/ensure-prg-seed.tact:4:16:
  3 |         // Bad: Calling nativeRandom without PRG init
> 4 |         return nativeRandom();
                     ^
  5 |     }
Help: Use `nativePrepareRandom`, `nativeRandomize`, `nativeRandomizeLt` to initialize the PRG seed or choose the safer `randomInt` function
See: https://nowarp.io/tools/misti/docs/detectors/EnsurePrgSeed

[INFO] PreferredStdlibApi: nativeRandom has a safer alternative: randomInt
test/good/ensure-prg-seed.tact:4:16:
  3 |         // Bad: Calling nativeRandom without PRG init
> 4 |         return nativeRandom();
                     ^
  5 |     }
Prefer `randomInt` since `nativeRandom` requires additional initialization of PRG before use
Help: nativeRandom should be replaced with a safer alternative: randomInt
See: https://nowarp.io/tools/misti/docs/detectors/PreferredStdlibApi